File: 55.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        reachable = 0
        for index, num in enumerate(nums):
            if(index > reachable):
                return False
            reachable = max(reachable, index + num)

        return reachable >= len(nums) - 1

File: 55.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int max_jump = 0;
        for(int i = 0; i < nums.size(); i++) {
            if(i > max_jump) {
                return false;
            }
            max_jump = fmax(max_jump, nums[i] + i);
        }
        return (max_jump >= nums.size() - 1);
    }
};